### ISSUE TITLE
StorageManager: Add x-amz-meta header

### ISSUE THREAD

**Issue by brian-visikon** [aws-amplify/amplify-ui#3683]
### On which framework/platform would you like to see this feature implemented?

React

### Which UI component is this feature-request for?

Storage (File Uploader)

### Please describe your feature-request in detail.

I would like to add metadata to the files I upload using the [StorageManager](https://ui.docs.amplify.aws/react/connected-components/storage/storagemanager) component, but it seems I cannot add custom metadata to the upload.

https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingMetadata.html#UserMetadata

### Please describe a solution you'd like.

Here is example of I would like to be able do:

```
<StorageManager
          processFile={({ file, key }) => {
            return { 
               file,  key, headers: [{ 'x-amz-meta-id': uuidv4()}] 
            };
            onUploadSuccess={({ key, headers }) => {
               setDone(true);
               console.log('Uploaded', key, headers);
            }}
          }}
        />
```

I obviously could add the ID to the filename in the `processFile()` callback, but this is not elegant and it quickly becomes problematic. Especially once I start passing more than just an ID. There could be contextual info also, such as a reference to the enabled feature flags.

### We love contributors! Is this something you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request.
- [ ] ⚠️ This feature might incur a breaking change.

**Comment by reesscot** [comment]
@brian-visikon You are correct that currently it is not possible to pass metadata, but I can see how this would be useful and it is supported by the underlying Amplify JS `Storage.put`. Are there any other options you would find useful from the "other options" section on this page? https://docs.amplify.aws/lib/storage/upload/q/platform/js/#encrypted-uploads

**Comment by brian-visikon** [comment]
Adding `x-amz-meta`  headers is the only thing I am missing.

**Comment by dbanksdesign** [comment]
@brian-visikon we merged a change that would allow this. It is currently published in the `@next` tag and will be fully released tomorrow. Here is an example:

https://github.com/aws-amplify/amplify-ui/blob/main/examples/next/pages/ui/components/storage/storage-manager/metadata/index.page.tsx

**Comment by dbanksdesign** [comment]
This was released in v[1.2.0](https://www.npmjs.com/package/@aws-amplify/ui-react-storage/v/1.2.0)!
